### ISSUE TITLE
this readme just went off a cliff

### ISSUE THREAD

**Issue by bhollan** [learn-co-curriculum/angular-sharing-directive-controllers-readme#4]
OK, I get it.  Todd Motto is the shit.  Sure.

But this is nuts.  I'm over 50 lessons in and I have _NO_ idea how to debug _ANYTHING_ we've learned so far.  It either works or it doesn't.  Angular is _just magical_.

To say that I'm 'angry' or 'upset' isn't really on point.  I'm confused.  We've done over 400 other lessons where we get to 'read the manual', then maybe 'watch someone else do it', and then 'see if you can do it on your own.'  These Angular lessons seem to be 'here, this is how 10,000 other developers have said things should be' and then 'now regurgitate it back'.  It's like one teacher gives us our own toolbox to work on projects ourselves, but then another teacher has us memorize stuff and gives us a written quiz.  It's a discontinuity like I never thought possible.

This whole track needs help.  It doesn't fit with the rest of the material at all.

For example: a few lessons ago (it was something about Directive Scope) we talked in grave detail about 4 ways to do something, 2 of which are never done.  The Directive Restrict was fine: "here are these other settings and what they mean, but they shouldn't really event exist."

At my last job we had a problem: everybody was always more interested in when things went _right_.  It's SO much more important and helpful to know how to _fix_ things when they go _wrong_.  These lessons don't even seem to be teaching a style or terminology, just "make it look like this, and the tests will pass."

I also don't think the coffee shop analogy fits at all.  That fits perfectly for the server because people are coming in with requests.  I think a better analogy would be a fashion show (maybe a broadway play?).  At a fashion show, people come to look, but you have gophers who go out to get things.  You have a stage manager queuing up things to be displayed.  Anyway, I just think it's silly to use the coffee shop analogy for a server and a client-side MVC.  Nobody leaves a coffee shop to get coffee.

If I got to choose one thing to change up to this point in the Angular track it would be debugging.  How do I pause this giant Angular machine?  What should I see when I do?  How can I tell what I did wrong?  How do I play with it?  Being able to play with something is vital to learning how to use it properly.  Maybe have some labs spread throughout that have a deliberate error (or 3) that's breaking 4 of 45 tests or something.  That way you can see how it's all connected while you're fixing things.

This material seems to be on an entirely different level, and at an unhealthy/unhelpful/disparate contrast to the rest of Learn.  This material seems like it's written to developers, not to learners.  It's really hard for someone who's been doing something for 10+ years to show someone how to do it who has never done it before.  Imagine having never coded before in your life and trying to wrap your head around the different between $scope, $scope, scope, and scope.  Why in the world do you need so many damn controllers too?

paging @AnnJohn

**Comment by gj** [comment]
I agree generally with most of these sentiments/suggestions. The entire Angular section (to this point, which is over halfway through the 111-lesson suite) pales in comparison to the rest of the Learn curriculum. It's poorly-written, barely-explained, and, as @bhollan points out, sorely lacking in real, test-driven examples that drive home the concepts we're supposed to be learning. I suppose part of the problem is relative: if the Learn curriculum to this point hadn't set such a high bar, the Angular section's shortcomings wouldn't be quite so glaringly obvious.

I'm doing fine with the material so far, but that's more due to the plethora of external resources I've consulted (such as [this great explainer on scopes](https://github.com/angular/angular.js/wiki/Understanding-Scopes)) than any of the lessons on Learn.

cc: @AnnJohn @aviflombaum
